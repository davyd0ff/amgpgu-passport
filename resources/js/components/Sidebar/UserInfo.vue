<template>
  <div class="user-view">
    <div class="background">
      <img
        src="https://www.las.ru/media/foto/kabinet-rukovoditelya-seventies-foto-01-a.jpg"
      />
    </div>
    <router-link to="/profile">
      <img
        class="circle"
        src="https://sun9-11.userapi.com/impf/c638426/v638426062/50419/eNL3yaz6niM.jpg?size=300x300&quality=96&proxy=1&sign=281eca8b1027b3a552a1e591fd6d5019&type=audio"
      />
    </router-link>
    <router-link to="/profile">
      <span class="black-text name">{{ username }}</span>
    </router-link>
    <router-link to="/profile">
      <span class="black-text email">{{ email }}</span>
    </router-link>
    <logout-button />
  </div>
</template>

<script>
import LogoutButton from './LogoutButton';
export default {
  name: 'UserInfo',
  components: { LogoutButton },
  computed: {
    info: function () {
      return this.$store.getters.user;
    },
    username: function () {
      if (this.info && this.info.lastname) {
        return `${this.info.lastname} ${this.info.firstname} ${this.info.middlename}`;
      }
      return '';
    },
    email: function () {
      if (this.info && this.info.email) {
        return this.info.email;
      }
      return '';
    },
  },
};
</script>

<style scoped></style>
