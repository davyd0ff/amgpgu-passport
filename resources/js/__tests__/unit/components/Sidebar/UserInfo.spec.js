import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import UserInfo from '@/components/Sidebar/UserInfo.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('UserInfo.vue', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        user: {
          lastname: 'Testov',
          firstname: 'Test',
          middlename: 'Testovich',
          email: 'test@test.ru',
        },
      },
      getters: {
        user: (state) => state.user,
      },
    });
  });

  it('render', () => {
    const wrapper = shallowMount(UserInfo, {
      store,
      localVue,
      stubs: ['router-link'],
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
