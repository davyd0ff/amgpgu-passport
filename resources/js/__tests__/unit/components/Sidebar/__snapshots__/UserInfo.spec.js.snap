// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo.vue render 1`] = `
<div
  class="user-view"
>
  <div
    class="background"
  >
    <img
      src="https://www.las.ru/media/foto/kabinet-rukovoditelya-seventies-foto-01-a.jpg"
    />
  </div>
   
  <router-link-stub
    to="/profile"
  >
    <img
      class="circle"
      src="https://sun9-11.userapi.com/impf/c638426/v638426062/50419/eNL3yaz6niM.jpg?size=300x300&quality=96&proxy=1&sign=281eca8b1027b3a552a1e591fd6d5019&type=audio"
    />
  </router-link-stub>
   
  <router-link-stub
    to="/profile"
  >
    <span
      class="black-text name"
    >
      Testov Test Testovich
    </span>
  </router-link-stub>
   
  <router-link-stub
    to="/profile"
  >
    <span
      class="black-text email"
    >
      test@test.ru
    </span>
  </router-link-stub>
   
  <logout-button-stub />
</div>
`;
